src/modules.js

```diff
@@ -0,0 +1,450 @@
+                let proj = dataModel.allTasks.customProjects
+                for (let i=0; i< proj.length; i++) {
+                    if (proj[i].name == selected) {
+                        t.list = selected
+                    }
+                    localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks))
+                }
+                let currentDiv = buttonS.parentElement;
+                let parent = currentDiv.parentElement;
+                parent.removeChild(currentDiv)
+            };
+        };
+
+        const editTask =(img) => {
+            img.onclick = (e) => {
+                let index = img.parentElement.getAttribute('data-index');
+                DOM.createTask();
+                document.getElementById('title').value = dataModel.allTasks.tasks[index].title;
+                document.getElementById('notes').value = dataModel.allTasks.tasks[index].notes;
+                document.getElementById('dueDate').value = dataModel.allTasks.tasks[index].dueDate;
+                document.getElementById('priority').value = dataModel.allTasks.tasks[index].priority;
+                let main = document.querySelector('#main');
+                main.removeChild(img.parentElement);
+                dataModel.allTasks.tasks.splice(index, 1);
+                updateTaskIndex();
+                updateCardIndex();
+            }
+        }
+    
+        const cancelTaskToProjectHandler = (buttonC) => {
+            buttonC.onclick = (e) => {
+                let currentDiv = buttonC.parentElement;
+                let parent = currentDiv.parentElement;
+                parent.removeChild(currentDiv)
+            };
+        };
+
+    //project-based functions*******************************************************************
+
+    const saveNewProject = () => {
+        let project = {
+            "name": document.getElementById('projectName').value,
+        };
+        addProjectToStorage(project);
+        DOM.addProjectToList(project);
+    };
+
+    const addProjectToStorage = (project) => {
+        let list = JSON.parse(localStorage.getItem('allTasks')); 
+        if ( list == null) {
+            dataModel.allTasks.customProjects.push(project);
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks))
+        } else {
+            dataModel.allTasks.customProjects.push(project);
+            //localStorage.removeItem('allTasks');
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks));
+        };
+    };
+
+    const createAddToProject = (img) => {
+        img.onclick = (e) => {
+            let list = JSON.parse(localStorage.getItem('allTasks'));
+            list = list.customProjects
+            if (list.length == 0) {
+                let div = document.createElement('div');
+                div.classList.add('projectAddPopup');
+                div.innerText= `Create a Project first
+                
+                `;
```

Comment:
If you were looking for a way to add newlines to something you can use `\n` instead. So this would be:
```
div.innerText= 'Create a Project first\n\n';
```
A lot easier to read compared to whitespace too.